src/main/java/httpserver/HttpRequestParser.java

```diff
@@ -0,0 +1,21 @@
+package httpserver;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class HttpRequestParser {
+    public static HttpRequest fromString(String rawRequest){
+        List<String> chunkedRawRequest = Arrays.asList(rawRequest.split("\r\n"));
+        String[] statusLine = chunkedRawRequest.get(0).split(" ");
+        String body = "";
+
+        if(chunkedRawRequest.contains("")){
+            body = chunkedRawRequest.get(chunkedRawRequest.size() - 1);
+        }
+
+        return new HttpRequest.HttpRequestBuilder()
+                .addRoute(statusLine[1])
```

Comment:
What happens if there is nothing at index 1? Is that ok?

Reply:
I'm looking into matchers to get away from using arrays directly, but it's going to take a bit of research. I thought using regex and capture groups via the `matcher` class would work, but I ran into an issue with headers. For now, I'm going to just stick to arrays, but I will continue working on a way to refactor this better. Hopefully I'll find a way before the next IPM.